algebra/elliptic-curve/src/proptest.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{Affine, Jacobian, BETA};
+use proptest::prelude::*;
+use zkp_primefield::{FieldElement, Pow, SquareRoot};
+
+impl Arbitrary for Affine {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+        FieldElement::arbitrary()
+            .prop_filter_map("x not on curve", |x| {
+                let y = (x.pow(3_usize) + &x + BETA).square_root();
```

Comment:
This is missing half the points, in particular it's missing the points that would have a negative y, so has a noticeable bias. Should be easy to fix with something like:

```rust
        (FieldElement::arbitrary(), bool)
            .prop_filter_map("x not on curve", |(x, sign)| {
                let y = (x.pow(3_usize) + &x + BETA).square_root();
                let y = if sign { -y } else { y };
```